src/lib.rs

```diff
@@ -458,7 +458,17 @@ impl core::fmt::LowerHex for SecretKey {
 }
 
 impl Signature {
-    pub fn parse(p: &[u8; util::SIGNATURE_SIZE]) -> Signature {
+    /// Parse an possibly overflowing signature.
+    ///
+    /// A SECP256K1 signature is usually required to be within 0 and curve
+    /// order. This function, however, allows signatures larger than curve order
+    /// by taking the signature and minus curve order.
+    ///
+    /// Note that while this function is technically safe, it is non-standard,
+    /// meaning you will have compatibility issues if you also use other
+    /// SECP256K! libraries. It's not recommended to use this function. Please
+    /// use `parse_standard` instead.
+    pub fn parse_overflowing(p: &[u8; util::SIGNATURE_SIZE]) -> Signature {
```

Comment:
Any reason not to use `#[deprecated(…)]` here?

Reply:
`deprecated` imply that we can eventually remove this, which we cannot do. Same reason why `der-lax` is not marked deprecated.

Reply:
While that is true I’d probably use it anyway to get those nice compile time warnings. You could set the depreciation version very far out perhaps? 
Either way, your call.

Reply:
I just think we simply can't get rid of it. In Substrate we'll have to create a new function version of this, with the old one always exist (and we cannot remove). In this sense, `deprecated` warning is just noise and not really useful.

We changed the function names, so when people upgrade they'll have a compile failure anyway and have to change the function name, which is more than safe enough to make sure they don't use the incorrect function.